src/main/java/hudson/plugins/cobertura/renderers/SourceCodePainter.java

```diff
@@ -27,19 +28,19 @@
  * @author Stephen Connolly
  * @since 31-Aug-2007 16:52:25
  */
-public class SourceCodePainter implements FilePath.FileCallable<Boolean>, Serializable {
+public class SourceCodePainter extends SlaveToMasterFileCallable<Boolean> implements Serializable {
```

Comment:
Again, at least for the invocation from `CoberturaPublisher.perform`, this is in fact a `MasterToSlaveFileCallable`.

The call from `MavenCoberturaPublisher.postExecute` is harder to follow. I think it is run entirely inside the agent, or even the Maven process, in which case the role check is irrelevant. At any rate, the [default whitelist](https://github.com/jenkinsci/jenkins/blob/1868c8486bdb7a39a7beec0787bb40e15b80de48/core/src/main/resources/jenkins/security/s2m/callable.conf#L30) makes no mention of this plugin, so I am assuming it could not have been using implicit slave-to-master calls before.